cirq-core/cirq/ops/boolean_hamiltonian.py

```diff
@@ -72,6 +73,8 @@ def __init__(
     def with_qubits(self, *new_qubits: 'cirq.Qid') -> 'BooleanHamiltonian':
         if len(self._qubit_map) != len(new_qubits):
             raise ValueError('Length of replacement qubits must be the same')
+        if protocols.qid_shape(tuple(self._qubit_map.values())) != protocols.qid_shape(new_qubits):
```

Comment:
So the gate only has qubits (2-valued), but I am OK with having the extra check, of course.

Reply:
I see. I simplified this to only allow 2-value qubits, and moved the dimensionality check to the init method.